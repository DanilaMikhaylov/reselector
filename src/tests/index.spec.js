import React from 'react'
import { transformFileSync } from 'babel-core'
import { mount } from 'enzyme'

import {
  ClassComponent,
  ArrowFunctionalComponent,
  FunctionalComponent,
} from './App'
import { get } from '../'

describe('babel plugin', () => {
  it('test transformed source code', () => {
    const { code } = transformFileSync(require.resolve('./App'))

    expect(code).toMatchSnapshot()
  })

  it('should find all the types of Components', () => {
    const components = [
      ClassComponent,
      ArrowFunctionalComponent,
      FunctionalComponent,
    ]

    components.forEach((Component) => {
      const wrapper = mount(<div><Component /></div>)

      expect(wrapper).toMatchSnapshot()
      expect(wrapper.find(get`${Component}`).length).toBe(1)
    })
  })

  it('should find nested Component', () => {
    const wrapper = mount((
      <ClassComponent>
        <ArrowFunctionalComponent>
          <FunctionalComponent />
        </ArrowFunctionalComponent>
      </ClassComponent>
    ))

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find(
      get`${ClassComponent} ${ArrowFunctionalComponent} ${FunctionalComponent}`,
    ).length).toBe(1)
  })
})
