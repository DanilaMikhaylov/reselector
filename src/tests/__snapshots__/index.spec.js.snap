// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel plugin should find all the types of Components 1`] = `
<div>
  <ClassComponent>
    <div
      data-test="0"
    />
  </ClassComponent>
</div>
`;

exports[`babel plugin should find all the types of Components 2`] = `
<div>
  <Component>
    <div
      data-test="1"
    />
  </Component>
</div>
`;

exports[`babel plugin should find all the types of Components 3`] = `
<div>
  <FunctionalComponent>
    <p
      data-test="2"
    />
  </FunctionalComponent>
</div>
`;

exports[`babel plugin should find nested Component 1`] = `
<ClassComponent>
  <div
    data-test="0"
  >
    <Component>
      <div
        data-test="1"
      >
        <FunctionalComponent>
          <p
            data-test="2"
          />
        </FunctionalComponent>
      </div>
    </Component>
  </div>
</ClassComponent>
`;

exports[`babel plugin test transformed source code 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.ArrowFunctionalComponent = exports.ClassComponent = undefined;
exports.FunctionalComponent = FunctionalComponent;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

class ClassComponent extends _react2.default.Component {
  render() {
    return _react2.default.createElement(
      'div',
      {
        'data-test': '0'
      },
      this.props.children
    );
  }
}

exports.ClassComponent = ClassComponent; /* eslint-disable react/prefer-stateless-function */

ClassComponent.testId = 0;
const ArrowFunctionalComponent = exports.ArrowFunctionalComponent = ({ children }) => _react2.default.createElement(
  'div',
  {
    'data-test': '1'
  },
  children
);

ArrowFunctionalComponent.testId = 1;
function FunctionalComponent({ children }) {
  return _react2.default.createElement(
    'p',
    {
      'data-test': '2'
    },
    children
  );
}
FunctionalComponent.testId = 2;"
`;
